src/core/execute.c

```diff
@@ -4081,6 +4089,11 @@ static int apply_mount_namespace(
         else
                 ns_info = (NamespaceInfo) {};
 
+        /* Symlinks (exec dirs, os-release) are set up after other mounts, before they are made read-only. */
+        r = compile_symlinks(context, params, ns_info.mount_apivfs && (root_dir || root_image), &symlinks);
```

Comment:
`ns_info.mount_apivfs && (root_dir || root_image)` appears three times. Maybe define a local `const bool` variable to make it clear to the reader.

Reply:
added